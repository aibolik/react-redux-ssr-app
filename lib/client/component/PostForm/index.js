var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.scss';

let PostForm = function (_Component) {
  _inherits(PostForm, _Component);

  function PostForm(...args) {
    var _temp, _this, _ret;

    _classCallCheck(this, PostForm);

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (PostForm.__proto__ || Object.getPrototypeOf(PostForm)).call(this, ...args)), _this), _this.submitPost = () => {
      let author = 'No author';
      let text = _this.refs.text.value;

      _this.props.createPost({ author, text });
      _this.refs.text.value = '';
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(PostForm, [{
    key: 'render',
    value: function render() {

      return React.createElement(
        'div',
        { className: 'post-form' },
        React.createElement('textarea', { className: 'post-form__text', placeholder: 'Your text', ref: 'text' }),
        React.createElement(
          'button',
          { className: 'btn', onClick: this.submitPost },
          'Submit'
        )
      );
    }
  }]);

  return PostForm;
}(Component);

PostForm.propTypes = {
  sendPost: PropTypes.func
};


export default PostForm;