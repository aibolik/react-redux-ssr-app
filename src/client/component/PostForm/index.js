import React, { Component } from 'react';
import PropTypes from 'prop-types';

class PostForm extends Component {
  static propTypes = {
    sendPost: PropTypes.func
  };

  submitPost = (e) => {
    console.log(e.data)
  };
  
  render() {
    
    return (
      <div>
        <input type='text' name='author' placeholder='Author' />
        <textarea name='text' placeholder='Your post' />
        <button onClick={this.submitPost}>Submit</button>
      </div>
    );
  }
}

export default PostForm;