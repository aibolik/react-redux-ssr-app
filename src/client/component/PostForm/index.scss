@import '../styles/variables';

.post-form {
  margin: 20px 0;

  &__text {
    color: $font-color;
    background: transparent;
    border: none;
    transition: all .3s;

    &:focus {
      outline: none;
      border-bottom: 2px solid $font-color;

      &::placeholder {
        color: $font-color;
      }
    }

    &::placeholder {
      color: transparentize($font-color, 0.2);
    }
  }

  &__btn {
    display: block;
    margin: 12px 0;
    padding: 6px 12px;
    cursor: pointer;
    user-select: none;
    transition: all 150ms linear;
    text-align: center;
    background: transparent;
    color: $font-color;
    border: 2px solid $font-color;
    border-radius: 4px;

    font-size: 13px;
    font-weight: 500;
    line-height: 1.3;

    &:hover {
      transition: all 150ms linear;
      opacity: .85;
    }
  }
}
