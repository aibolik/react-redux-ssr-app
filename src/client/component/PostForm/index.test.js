import React from 'react';
import ReactDOM from 'react-dom';
import PostForm from './index';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestUtils from 'react-dom/test-utils';

describe('Test PostForm component', () => {
  const mockProps = {
    createPost: jest.fn(),
    author: {
      id: '5aa0dedae3fa01461ca2de06',
      name: 'Test Name'
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PostForm />).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
