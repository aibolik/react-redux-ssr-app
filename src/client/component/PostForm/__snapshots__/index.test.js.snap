// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PostForm component renders correctly 1`] = `
<div
  className="post-form"
>
  <textarea
    className="post-form__text"
    placeholder="Your text"
  />
  <button
    className="btn"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
